### PR TITLE
feat: add stealth relayer to jobs

### PR DESCRIPTION
**Pull request by lbertenasco**

### CODE REVIEW

File: contracts/jobs/v2/HarvestV2QueueKeep3rJob.sol [hunk 1]
```diff
@@ -19,7 +21,8 @@ contract HarvestV2QueueKeep3rJob is V2QueueKeep3rJob {
         public
         V2QueueKeep3rJob(
             _mechanicsRegistry,
-            address(0), /*TODO:_yOracle*/
+            _stealthRelayer,
+            _yOracle, /*TODO:_yOracle*/
```
Comment (alejoamiras) [inline contracts/jobs/v2/HarvestV2QueueKeep3rJob.sol:25]:
```suggestion
            _yOracle,
```

File: contracts/utils/OnlyStealthRelayer.sol [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    modifier onlyStealthRelayer() {
+        require(stealthRelayer == address(0) || msg.sender == stealthRelayer, "OnlyStealthRelayer::msg-sender-not-stealth-relayer");
+        _;
+    }
+
+    function _setStealthRelayer(address _stealthRelayer) internal {
+        stealthRelayer = _stealthRelayer;
+        emit StealthRelayerSet(_stealthRelayer);
+    }
+}
```
Comment (alejoamiras) [inline contracts/utils/OnlyStealthRelayer.sol:32]: Maybe rename to `StealthRelayerSupported` or something like that ? Either way, not a big thing 😋

File: contracts/utils/OnlyStealthRelayer.sol [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ */
+abstract contract OnlyStealthRelayer is IOnlyStealthRelayer {
+    address public override stealthRelayer;
+
+    constructor(address _stealthRelayer) public {
+        _setStealthRelayer(_stealthRelayer);
+    }
+
+    modifier onlyStealthRelayer() {
+        require(stealthRelayer == address(0) || msg.sender == stealthRelayer, "OnlyStealthRelayer::msg-sender-not-stealth-relayer");
```
Comment (alejoamiras) [inline contracts/utils/OnlyStealthRelayer.sol:24]:
```suggestion
        require(stealthRelayer == address(0) || msg.sender == stealthRelayer, "OnlyStealthRelayer::not-realyer");
```
Ive done some research, and if require messages are bigger than 32 bytes they take 32 more bytes. So, in order to make things cheaper, we need to shorten our messages 😭

**Review by alejoamiras — APPROVED** [review]
I need to re-read stealth tx implementation, but this looks clean, no obvious bugs or attacks.

File: contracts/utils/OnlyStealthRelayer.sol (hunk 3 above)
Comment (lbertenasco) [reply contracts/utils/OnlyStealthRelayer.sol]: nice catch!

File: contracts/jobs/v2/HarvestV2QueueKeep3rJob.sol (hunk 1 above)
Comment (lbertenasco) [reply contracts/jobs/v2/HarvestV2QueueKeep3rJob.sol:25]: still has no _yOracle support. this is why I left todo on code

**Pull request merged**